Cognitive Services 2025-07-01-preview

CarlOchs opened the pull request:
# ARM (Control Plane) API Specification Update Pull Request 

> [!TIP]
> Overwhelmed by all this guidance? See the `Getting help` section at the bottom of this PR description.

## PR review workflow diagram

Please understand this diagram before proceeding. It explains how to get your PR approved & merged.

![spec_pr_review_workflow_diagram](https://github.com/Azure/azure-rest-api-specs/assets/4429827/5bb5e7ce-8aff-4dbb-a3f8-0d9b68fef5b1)

## Purpose of this PR

What's the purpose of this PR? Check the specific option that applies. This is **mandatory**!

  - [ ] New resource provider.
  - [X] New API version for an existing resource provider. (If API spec is not defined in TypeSpec, the PR should have been created in adherence to [OpenAPI specs PR creation guidance](https://aka.ms/azsdkdocs/createopenapispec)).
  - [ ] Update existing version for a new feature. (This is applicable only when you are revising a private preview API version.)
  - [ ] Update existing version to fix OpenAPI spec quality issues in S360.
  - [ ] Convert existing [OpenAPI spec to TypeSpec spec](https://aka.ms/typespec/conversion) (do not combine this with implementing changes for a new API version).
  - [ ] Other, please clarify:
    - _edit this with your clarification_

## Due diligence checklist

To merge this PR, you **must** go through the following checklist and confirm you understood 
and followed the instructions by checking all the boxes:

- [X] I confirm this PR is modifying Azure Resource Manager (ARM) related specifications, and not data plane related specifications.
- [X] I have reviewed following [Resource Provider guidelines](https://aka.ms/rpguidelines), including
  [ARM resource provider contract](https://aka.ms/azurerpc) and
  [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md) (estimated time: 4 hours).  
  I understand this is required before I can proceed to the diagram Step 2, "ARM API changes review", for this PR.
- [X] A [release plan](https://aka.ms/azsdkdocs/release-plans) has been created. If not, please create one as it will help guide you through the REST API and SDK creation process. 

## Additional information

<details>
<summary> Viewing API changes</summary>

For convenient view of the API changes made by this PR, refer to the URLs provided in the table 
in the `Generated ApiView` comment added to this PR. You can use ApiView to show API versions diff. 

</details>
<details>
<summary>Suppressing failures</summary>

If one or multiple validation error/warning suppression(s) is detected in your PR, please follow the 
[suppressions guide](https://aka.ms/azsdk/pr-suppressions) to get approval.

</details>

## Getting help

- First, please carefully read through this PR description, from top to bottom. Please fill out the `Purpose of this PR` and `Due diligence checklist`.
- If you don't have permissions to remove or add labels to the PR, request `write access` per [aka.ms/azsdk/access#request-access-to-rest-api-or-sdk-repositories](https://aka.ms/azsdk/access#request-access-to-rest-api-or-sdk-repositories)
- To understand what you must do next to merge this PR, see the `Next Steps to Merge` comment. It will appear within few minutes of submitting this PR and will continue to be up-to-date with current PR state.
- For guidance on fixing this PR CI check failures, see the hyperlinks provided in given failure 
  and https://aka.ms/ci-fix.
- For help with ARM review (PR workflow diagram Step 2), see https://aka.ms/azsdk/pr-arm-review.
- If the PR CI checks appear to be stuck in `queued` state, please add a comment with contents `/azp run`.
  This should result in a new comment denoting a `PR validation pipeline` has started and the checks should be updated after few minutes.
- If the help provided by the previous points is not enough, post to https://aka.ms/azsdk/support/specreview-channel and link to this PR.
- For guidance on SDK breaking change review, refer to https://aka.ms/ci-fix.

CarlOchs committed:
Baseline against 2025-06-01 stable.

CarlOchs commented:
AzSDK Release Plan: https://aka.ms/sdk-release-planner?release-plan-id=2f2fb217-614c-f011-8779-00224804b6c3

rkmanda commented:
>     "https"

 When introducing a new API version, create the folder for the new API version and place files from the previous latest API version as is in the first commit. All new changes to the json files that you would like to ship with this new API version you are introducing should only be added in the subsequent commits. 
 This allows the reviewer to use the diffing capabilities between the commits of a PR in tools like codeflow to get a clear understanding of the actual changes being introduced. With the way the PR is raised now, the reviewer cannot discover what the new changes are. Please either abandon the PR and raise another one with the recommendation or create a new set of commits on this PR following the recommendation. If you have already followed this pattern in this PR, p[lease indicate which commit is the exact copy of the previous version.


---
Refers to: specification/cognitiveservices/resource-manager/Microsoft.CognitiveServices/preview/2025-07-01-preview/cognitiveservices.json:10 in 8b40f65. [](commit_id = 8b40f65ab19f550a4105ce44d7b416bed4961f2b, deletion_comment = False)

rkmanda commented:
>     "https"

There is just a single commit in this PR so its hard to tell what the diffs being introduced are

---
In reply to: [2986583396](https://github.com/Azure/azure-rest-api-specs/pull/35345#issuecomment-2986583396) [](http://example.com/codeflow?ancestors=2986583396)

---
Refers to: specification/cognitiveservices/resource-manager/Microsoft.CognitiveServices/preview/2025-07-01-preview/cognitiveservices.json:10 in 8b40f65. [](commit_id = 8b40f65ab19f550a4105ce44d7b416bed4961f2b, deletion_comment = False)

CarlOchs committed:
Update readme.md to use package-2025-07-01-preview by default

ritwikdesai committed:
Added Quota Tier Swagger Specs  (#35388)

* added GET/PATCH specs for quota tier
* defaultParameter fix
* examples fix
* model fix
* fixed eligibility contract

huliang-microsoft committed:
[CustomTopic] Add custom topic feature to cogsvc control plane API (#35372)

* Add custom topic feature paths
* Add examples and fix bugs
* Fix pipeline

aditya30394 committed:
Added new Legacy lifecycle status, replacement config and model asset id (#35505)

* Added new Legacy lifecycle status with detailed enum documentation, introduced ReplacementConfig object for automated model migration guidance, and added modelCatalogAssetId for model catalog information

* Spell check and version change in examples

aditya30394 committed:
ReplacementConfig.autoUpgradeStartDate should be in format of date-time (#35655)

* ReplacementConfig.autoUpgradeStartDate should be in format of date-time
* Update the examples too

CarlOchs committed:
Merge branch 'main' into carlochs/cogsvc-2025-07-01-preview

CarlOchs committed:
Merge branch 'main' into carlochs/cogsvc-2025-07-01-preview

chnldw committed:
Add storedCompletionsDisabled Account Properties: The flag to disable stored completions. (#35697)

Co-authored-by: Dawei Li <daweil@microsoft.com>

psah434 commented:
Please fix the comments and what rmanda has requested changes for.

CarlOchs commented:
> Please fix the comments and what rmanda has requested changes for.

The initial commit to this PR used the 2025-06-01 stable as the baseline.  Should I have used the last preview version (2025-04-01-preview)?  

I did make a mistake in opening the PR with the initial commit and not flagging it as "not-ready-for-review" (though I'm not sure the best way to do that), and your team is on top of these PRs.

CarlOchs committed:
Merge branch 'main' into carlochs/cogsvc-2025-07-01-preview

gary-x-li commented:
>               "$ref": "../../../../../common-types/resource-management/v2/types.json#/definitions/ErrorResponse"

Trying using the latest version (v5 or v6) for common types. I think the minimal version should be v3.

---
Refers to: specification/cognitiveservices/resource-manager/Microsoft.CognitiveServices/preview/2025-07-01-preview/cognitiveservices.json:2770 in 9c65d9c. [](commit_id = 9c65d9c9e5253312461945ccffd3ea1b88803a01, deletion_comment = False)

gary-x-li commented:
>         "storedCompletionsDisabled": {

Consider changing it to enum so that you can add more future values instead of adding more boolean properties.

More context on the topic:
ARM recommends enums over booleans for future proof APIs. Standard guidance is: replace boolean/switch properties with a more meaningful enum whenever possible.

A boolean will forever have two valid values (true or false). A string enum type is always preferred. Also, properties should always provide better values just than True and False. For example two switches "isTypeA" and "isTypeB" should be replaced with one enum "type": [A, B, DefaultType]. Enums are always a more flexible and future proof option because they allow additional values to be added in the future in a non-breaking way, e.g. [Enabled, Disabled, Suspended, Deallocated]. 

Note: do NOT define a 'boolean enum' with two values 'True and False'. This might be easier to 'extend' in terms of types, but semantically its confusing, and no better than a boolean. 

---
Refers to: specification/cognitiveservices/resource-manager/Microsoft.CognitiveServices/preview/2025-07-01-preview/cognitiveservices.json:5446 in 9c65d9c. [](commit_id = 9c65d9c9e5253312461945ccffd3ea1b88803a01, deletion_comment = False)

gary-x-li commented:
>         "blocking": {

Same as above. Consider using enum.

---
Refers to: specification/cognitiveservices/resource-manager/Microsoft.CognitiveServices/preview/2025-07-01-preview/cognitiveservices.json:8085 in 9c65d9c. [](commit_id = 9c65d9c9e5253312461945ccffd3ea1b88803a01, deletion_comment = False)

aditya30394 committed:
[Address PR feedback] Use enum instead of a boolean value to indicate deprecation status (#36189)

aditya30394 commented:
> isUpgradeAvailable

Addressed the feedback

ritwikdesai committed:
Added PUT/LIST operation for Quota Tiers (#36141)

* Added PUT operation for QuotaTiers

* updated quota tiers common types to v6

* added 201 code for quota tiers PUT

* revert common types to v2

* typo

* added list quota tiers api

* list fix

* nextLink added

* test

* test

* typo

* changed isUpgradeAvailable to upgradeAvailabilityStatus enum

* path revert

ritwikdesai commented:
> > ```
> >     "isUpgradeAvailable": {
> > ```
> 
> Same as above. Consider using enum.
> 
> Refers to: specification/cognitiveservices/resource-manager/Microsoft.CognitiveServices/preview/2025-07-01-preview/cognitiveservices.json:9506 in [9c65d9c](https://github.com/Azure/azure-rest-api-specs/commit/9c65d9c9e5253312461945ccffd3ea1b88803a01). [](commit_id = [9c65d9c](https://github.com/Azure/azure-rest-api-specs/commit/9c65d9c9e5253312461945ccffd3ea1b88803a01), deletion_comment = False)

@gary-x-li addressed

CarlOchs committed:
Update to reference newer common types (#36208)

* Update all common type references to use v3 - v4+ will require breaking change verification.
* Fix parameter format for Get/Update QuotaTier examples
* Enable modelerfour.lenient-model-deduplication for python and go SDK generation - ErrorResponse/ErrorDetail common types are being flagged without this setting.

CarlOchs committed:
Fix invalid NetworkInjections specification - bug since 2025-04-01-preview (#36376)

Within the Account properties, the networkInjections property needed to be an array of NetworkInjection objects.

hgarg11 committed:
Moderation List API changes (#36484)

* Moderation List API changes

* Fix semantic validation

* Added examples for ModerationList operations

* Added bulk operations for Moderation

* Adressed review comments

* Fixing example mismatches 1

* Fix linting errors

* Fixing issues

* Fixed another error

* Fixed another error

CarlOchs committed:
Revert "Moderation List API changes (#36484)" - Not ready for public usage.

This reverts commit 5f61322848db4b54d1e8638d868afc840eb5cb63.

rkmanda reviewed (approved):
:shipit:

rkuthala commented on specification/cognitiveservices/resource-manager/Microsoft.CognitiveServices/preview/2025-07-01-preview/cognitiveservices.json:
```diff
@@ -0,0 +1,9877 @@
+            }
+          },
+          "201": {
+            "description": "If resource is created successfully, the service should return 201 (OK).",
```
201 - Created

rkuthala commented on specification/cognitiveservices/resource-manager/Microsoft.CognitiveServices/preview/2025-07-01-preview/cognitiveservices.json:
```diff
@@ -0,0 +1,9877 @@
+            "schema": {
+              "$ref": "#/definitions/Account"
+            },
+            "description": "The parameters to provide for the created account."
```
should be `for the updated account`?

rkuthala commented on specification/cognitiveservices/resource-manager/Microsoft.CognitiveServices/preview/2025-07-01-preview/cognitiveservices.json:
```diff
@@ -0,0 +1,9877 @@
+        "tags": [
+          "CognitiveServicesAccounts"
+        ],
+        "description": "Deletes a Cognitive Services account from the resource group. ",
```
should we say it is Purging the account?

rkuthala commented on specification/cognitiveservices/resource-manager/Microsoft.CognitiveServices/preview/2025-07-01-preview/cognitiveservices.json:
```diff
@@ -0,0 +1,9877 @@
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.CognitiveServices/accounts/{accountName}/raiBlocklists/{raiBlocklistName}/deleteRaiBlocklistItems": {
+      "post": {
```
is this `delete` operation?

rkuthala commented on specification/cognitiveservices/resource-manager/Microsoft.CognitiveServices/preview/2025-07-01-preview/cognitiveservices.json:
```diff
@@ -0,0 +1,9877 @@
+        ],
+        "responses": {
+          "204": {
+            "description": "Added the blocklist Items successfully."
```
`s\Added\Deleted`

rkuthala commented on specification/cognitiveservices/resource-manager/Microsoft.CognitiveServices/preview/2025-07-01-preview/cognitiveservices.json:
```diff
@@ -0,0 +1,9877 @@
+            }
+          },
+          "201": {
+            "description": "If resource is created successfully, the service should return 201 (OK).",
```
Created

rkuthala commented on specification/cognitiveservices/resource-manager/Microsoft.CognitiveServices/preview/2025-07-01-preview/cognitiveservices.json:
```diff
@@ -0,0 +1,9877 @@
+                "type": "string"
+              }
+            },
+            "description": "If resource is updated successfully, the service should return 201 (OK)."
```
202

rkuthala commented on specification/cognitiveservices/resource-manager/Microsoft.CognitiveServices/preview/2025-07-01-preview/cognitiveservices.json:
```diff
@@ -0,0 +1,9877 @@
+            }
+          },
+          "201": {
+            "description": "If resource is created successfully, the service should return 201 (OK).",
```
Created

rkuthala reviewed (approved):
offline conversation - requested changes will be addressed in October preview
